src/ucp/proto/proto_multi.c

```diff
@@ -158,21 +101,37 @@ ucs_status_t ucp_proto_multi_init(const ucp_proto_multi_init_params_t *params,
         perf.latency             = ucs_max(perf.latency, lane_perf->latency);
         perf.sys_latency         = ucs_max(perf.sys_latency,
                                            lane_perf->sys_latency);
-        lane_map                |= UCS_BIT(lane);
     }
 
-    /* Initialize multi-lane private data and relative weights */
-    reg_md_map          = ucp_proto_common_reg_md_map(&params->super, lane_map);
-    mpriv->reg_md_map   = reg_md_map | params->initial_reg_md_map;
-    mpriv->lane_map     = lane_map;
-    mpriv->num_lanes    = 0;
-    mpriv->min_frag     = 0;
-    mpriv->max_frag_sum = 0;
-    mpriv->align_thresh = 1;
-    perf.max_frag       = 0;
-    perf.min_length     = 0;
-    weight_sum          = 0;
-    min_end_offset      = 0;
+    return perf;
+}
+
+static void
+ucp_proto_multi_init_priv(const ucp_proto_multi_init_params_t *params,
+                          ucp_proto_common_tl_perf_t *lanes_perf,
+                          ucp_lane_map_t lane_map, ucp_md_map_t reg_md_map,
+                          double max_frag_ratio, size_t min_rndv_chunk_size,
+                          double min_bandwidth,
+                          ucp_proto_common_tl_perf_t *acc_perf,
+                          ucp_proto_multi_priv_t *mpriv)
+{
+    ucp_proto_common_tl_perf_t *lane_perf;
+    ucp_proto_multi_lane_priv_t *lpriv;
+    ucp_lane_index_t lane;
+    size_t min_length     = 0;
```

Comment:
code style: normally variables with initialization should come first

Reply:
Done, thanks